apps/aecore/src/aec_hard_forks.erl

```diff
@@ -156,3 +250,51 @@ is_valid_next_protocol({CurV, CurH}, {PrevV, PrevH}) when CurV > PrevV,
     true;
 is_valid_next_protocol(_, _) ->
     false.
+
+assert_fork(<<"ae_mainnet">>, Config, _Protocols) ->
+    assert_only_enabled_key(Config);
+assert_fork(<<"ae_uat">>, Config, _Protocols) ->
+    assert_only_enabled_key(Config);
+assert_fork(_Id, #{enabled := _Enabled,
+                   signalling_start_height := SignallingStartHeight,
+                   signalling_end_height := SignallingEndHeight,
+                   signalling_block_count := SignallingBlockCount,
+                   info_field := _InfoField,
+                   version := Version}, Protocols) ->
+    SortedProtocols = protocols_sorted_by_version(Protocols),
+    {PrevVersion, PrevForkHeight} = lists:last(SortedProtocols),
+    assert_fork_signalling_interval(SignallingStartHeight, SignallingEndHeight, PrevForkHeight),
+    SignallingInterval = SignallingEndHeight - SignallingStartHeight,
+    assert_fork_signalling_block_count(SignallingInterval, SignallingBlockCount),
+    assert_fork_version(Version, PrevVersion);
+assert_fork(_Id, undefined, _Protocols) ->
+    ok.
+
+assert_only_enabled_key(Config) when Config =/= undefined ->
+    %% ae_mainnet and ae_uat are allowed to have just 'enabled' key in the config.
+    case maps:size(maps:without([enabled], Config)) of
+        0      -> ok;
+        _Other -> error(illegal_fork_signalling_config)
+    end;
+assert_only_enabled_key(undefined) ->
+    ok.
+
+assert_fork_signalling_interval(SignallingStartHeight, SignallingEndHeight, PrevForkHeight)
+  when (PrevForkHeight < SignallingStartHeight) andalso
+       (SignallingStartHeight < SignallingEndHeight) ->
+    ok;
+assert_fork_signalling_interval(SignallingStartHeight, SignallingEndHeight, _PrevForkHeight) ->
+    error({illegal_fork_signalling_interval, SignallingStartHeight, SignallingEndHeight}).
+
+assert_fork_signalling_block_count(SignallingInterval, SignallingBlockCount)
+  when SignallingBlockCount =< SignallingInterval ->
+    ok;
+assert_fork_signalling_block_count(_SignallingInterval, SignallingBlockCount) ->
+    error({illegal_fork_signalling_block_count, SignallingBlockCount}).
+
+assert_fork_version(Version, PrevVersion)
+  when (Version > ?MINERVA_PROTOCOL_VSN) andalso
```

Comment:
Why Minerva?

Reply:
Because info field is present after Minerva so it's possible to do community forks after Minerva in environments other than mainnet and testnet.